client/dashboard/components/responsive-menu/index.tsx

```diff
@@ -1,28 +1,61 @@
 import { __experimentalHStack as HStack, Button, DropdownMenu } from '@wordpress/components';
-import { useViewportMatch } from '@wordpress/compose';
 import { __ } from '@wordpress/i18n';
 import { menu } from '@wordpress/icons';
-import React, { type ComponentProps } from 'react';
+import React, { useEffect, useRef, useState, type ComponentProps } from 'react';
 import Menu from '../menu';
 import RouterLinkMenuItem from '../router-link-menu-item';
 
 type ResponsiveMenuProps = {
+	prefix?: React.ReactNode;
 	children: React.ReactNode;
 	icon?: React.ReactElement;
 	label?: string;
 	dropdownPlacement?: 'bottom-end' | 'bottom-start' | 'bottom';
+	forceCollapsed?: boolean;
+	onCollapseChange?: ( collapsed: boolean ) => void;
 };
 
 function ResponsiveMenu( {
+	prefix,
 	children,
 	icon = menu,
 	label = 'Menu',
 	dropdownPlacement = 'bottom-end',
+	forceCollapsed,
+	onCollapseChange,
 }: ResponsiveMenuProps ) {
-	const isDesktop = useViewportMatch( 'medium' );
+	const wrapperRef = useRef< HTMLDivElement | null >( null );
+	const measureRef = useRef< HTMLDivElement | null >( null );
+	const [ isCollapsed, setIsCollapsed ] = useState( false );
 
-	if ( isDesktop ) {
-		return (
+	const checkOverflow = () => {
```

Comment:
Maybe we can create a hook like [useTrackOverflow](https://github.com/WordPress/gutenberg/blob/416d16ceba57fd1f050f311d9fa2d550e222c642/packages/components/src/tabs/use-track-overflow.ts#L17) to track whether it overflows 🤔